scripts/Dockerfile.singularity-shim

```diff
@@ -10,9 +12,9 @@ RUN printf '#!/bin/bash\n\
                 GROUPNAME=host-user\n\
                 addgroup -g $GID host-user\n\
             fi\n\
-            adduser -u $UID -G "$GROUPNAME" -H -D host-user\n\
+            adduser -u $UID -G "$GROUPNAME" -D host-user\n\
```

Comment:
what was -H for originally?

Reply:
@yarikoptic 
The -H switch tells adduser to not make a home directory. The thought was to have minimal impact on the file system. However, v3.3.0 of Singularity wants to create a ~/.singularity directory and is demanding a home directory for the user exist.

Reply:
@yarikoptic 
The slim release simply has all the "convenience" packages removed. They are both Alpine 3.10.1, but slim has had basic packages such as bash and sudo removed. Slim is absolutely just the essential packages necessary to get singularity working.

Reply:
Thanks for the explanations!
re image & apk -- please add `&& rm -rf /var/cache/apk/*` at the end that `apk update` so we strip away packages lists etc in that layer